packages/catalog-process/test/catalogService.test.ts

```diff
@@ -1,1353 +0,0 @@
-          authData,
-          eService: addMetadata(eService),
-        })
-      ).toThrowError(eServiceDescriptorNotFound(eService.id, descriptorId));
-    });
-
-    it("returns an error if the eservice updated descriptor is not in a Draft state", async () => {
-      const descriptor: Descriptor = { ...mockDescriptor, state: "Published" };
-      const eService: EService = {
-        ...mockEservice,
-        descriptors: [descriptor],
-      };
-      expect(() =>
-        updateDescriptorLogic({
-          eserviceId: eService.id,
-          descriptorId: eService.descriptors[0].id,
-          seed: mockEserviceDescriptorSeed,
-          authData,
-          eService: addMetadata(eService),
-        })
-      ).toThrowError(
-        notValidDescriptor(eService.descriptors[0].id, "Published")
-      );
-    });
-
-    it("returns an error if the authenticated organization is not the producer", async () => {
-      expect(() =>
-        updateDescriptorLogic({
-          eserviceId: mockEservice.id,
-          descriptorId: mockEservice.descriptors[0].id,
-          seed: mockEserviceDescriptorSeed,
-          authData: {
-            ...authData,
-            organizationId: "other-org-id",
-          },
-          eService: addMetadata({
-            ...mockEservice,
-            producerId: "some-org-id",
-          }),
-        })
-      ).toThrowError(operationForbidden);
-    });
-
-    it("returns an error when the service does not exist", async () => {
-      const eserviceId = unsafeBrandId<EServiceId>("not-existing-id");
-      expect(() =>
-        updateDescriptorLogic({
-          eserviceId,
-          descriptorId: mockEservice.descriptors[0].id,
-          seed: mockEserviceDescriptorSeed,
-          authData: {
-            ...authData,
-            organizationId: "organizationId",
-          },
-          eService: undefined,
-        })
-      ).toThrowError(eServiceNotFound(eserviceId));
-    });
-  });
-  describe("publishDescriptor", () => {
-    it("publish the descriptor", async () => {
-      const eService: EService = {
-        ...mockEservice,
-        descriptors: [{ ...mockDescriptor, state: "Draft" }],
-      };
-      const event = publishDescriptorLogic({
-        eserviceId: eService.id,
-        descriptorId: eService.descriptors[0].id,
-        authData,
-        eService: addMetadata(eService),
-      });
-      expect(event[0].event.type).toBe("EServiceDescriptorUpdated");
-      expect(event[0].event.data).toMatchObject({
-        eServiceId: eService.id,
-        eServiceDescriptor: {
-          ...toDescriptorV1(mockDescriptor),
-          publishedAt: (
-            event[0].event.data as { eServiceDescriptor: { publishedAt: Date } }
-          ).eServiceDescriptor.publishedAt,
-          state: toEServiceDescriptorStateV1("Published"),
-        },
-      });
-    });
-
-    it("publish the descriptor and deprecate the current published descriptor", async () => {
```

Comment:
I think the only test that is missing in the other test suite (#156) is this one: we should check if the publication of the descriptor leads to the deprecation of the previous one. I can add this test case.